src/category_theory/mittag_leffler.lean

```diff
@@ -0,0 +1,278 @@
+/-
+Copyright (c) 2022 Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémi Bottinelli
+-/
+import category_theory.filtered
+import topology.category.Top.limits
+import data.finset.basic
+import category_theory.category.basic
+import category_theory.full_subcategory
+import data.set.finite
+import data.fintype.basic
+import category_theory.types
+
+/-!
+# Mittag Leffler
+
+This files defines the Mittag-Leffler condition for cofiltered systems and (TODO) other properties
+of such systems and their sections.
+
+## Main definitions
+
+Given the functor `F : J ⥤ Type v`:
+
+* For `j : J`, `F.eventual_range j` is the intersections of all ranges of morphisms `F.map f`
+  where `f` has codomain `j`.
+* `is_mittag_leffler` states that the functor `F : J → Type v`, satisfies the Mittag-Leffler
+  condition: the ranges of morphisms `F.map f` (with `f` having codomain `j`) stabilize.
+* If `J` is cofiltered `F.to_eventual_ranges` is the subfunctor of `F` obtained by restriction
+  to `F.eventual_range`.
+
+
+## Main statements
+
+* `is_mittag_leffler_of_exists_finite_range` shows that if `J` is cofiltered and for all `j`,
+  there exists some `i` and `f : i ⟶ j` such that the range of `F.map f` is finite, then
+  `F` is Mittag-Leffler.
+* `to_eventual_ranges_surjective` shows that if `F` is Mittag-Leffler, then `F.to_eventual_ranges`
+  has all morphisms `F.map f` surjective.
+
+## Todo
+
+* Specialize to inverse systems and fintype systems.
+* Prove [Stacks: Lemma 0597](https://stacks.math.columbia.edu/tag/0597)
+
+## References
+
+* [Stacks: Mittag-Leffler systems](https://stacks.math.columbia.edu/tag/0594)
+
+## Tags
+
+Mittag-Leffler, surjective, eventual range, inverse system,
+
+-/
+
+
+universes u v w
+
+namespace category_theory
+namespace functor
+
```

Comment:
I may have more comments to do later, but you could add `variables {J : Type u} [category J] (F : J ⥤ Type v)` here and remove these parameters in the subsequent defininitions.

Reply:
Will do, thanks!